cors_test.go

```diff
@@ -282,6 +282,9 @@ func TestPassesAllowOrigins(t *testing.T) {
 		AllowOriginFunc: func(origin string) bool {
 			return origin == "http://github.com"
 		},
+		AllowOriginWithContextFunc: func(c *gin.Context, origin string) bool {
+			return origin == "http://sample.com"
+		},
```

Comment:
Can you add the testing for your scenario?

```
Example

api.example.com/app1 only allows origins app1.example.com
api.example.com/app2 only allows origins from app2.example.com
api.example.com/v1 allows all origins

This would not be possible in the current cors configuration options.
```

Reply:
cc @dbhoot Please help to add more test case.

Reply:
@appleboy Added